lib/ocpp/v201/evse_manager.cpp

```diff
@@ -30,21 +30,21 @@ EvseManager::EvseIterator EvseManager::end() {
 }
 
 EvseInterface& EvseManager::get_evse(int32_t id) {
-    if (id == 0 or id > this->evses.size()) {
+    if (id <= 0 or id > this->evses.size()) {
         throw EvseOutOfRangeException(id);
     }
     return *this->evses.at(id - 1);
 }
 
-const EvseInterface& EvseManager::get_evse(int32_t id) const {
-    if (id == 0 or id > this->evses.size()) {
+const EvseInterface& EvseManager::get_evse(const int32_t id) const {
+    if (id <= 0 or id > this->evses.size()) {
         throw EvseOutOfRangeException(id);
     }
     return *this->evses.at(id - 1);
 }
 
-bool EvseManager::does_evse_exist(int32_t id) const {
```

Comment:
The name of this function is a bit strange. In principle, evse id 0 does not exist so I added that check. But composite schedule relies on this that evse id 0 does exist. So maybe we should change the name or add an extra parameter here? @Pietfried ?

Reply:
The fact that with smart charging profiles they (mis)use evse 0 should be covered in that part of the code I think. Now you can ask the EvseManager if evse 0 exists and it will say yes. Then please give me evse 0 and it will throw :thinking:

Reply:
Exactly, that's strange isn't it?
So it's not for this pull request but shall I make a ticket maybe?

Reply:
Well you could change this to be `id > 0 && etc` and then fix the issues in smart charging as well, should be a small fix. Or indeed in a different ticket if you prefer that.

Reply:
Yes I created a bug report, as the function is also called from another function and I think to be sure, we have to test smart charging if everything is still fine. 

https://github.com/EVerest/libocpp/issues/868